hal/src/mrc/include/mrc/NetComm.h

```diff
@@ -16,56 +18,94 @@
 
 namespace mrc {
 
-struct OpModeTrace {
-  uint64_t Hash{0};
-  bool Enabled{false};
-  bool NoCurrentOpMpode{false};
-
-  static OpModeTrace FromValue(uint64_t Value) {
-    return OpModeTrace{
-        .Hash = Value & 0x3FFFFFFFFFFFFFFF,
-        .Enabled = (Value & 0x8000000000000000) == 0,
-        .NoCurrentOpMpode = (Value & 0x4000000000000000) != 0,
-    };
+struct OpModeHash {
+  uint64_t Hash : 56 {};
+  uint64_t IsAuto : 1 {};
+  uint64_t IsTest : 1 {};
+  uint64_t IsEnabled : 1 {};
+  uint64_t Reserved : 6 {};
+
+  static OpModeHash MakeTest(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsTest = 1;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash MakeTele(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash MakeAuto(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsAuto = 1;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash FromValue(uint64_t Value) {
+    OpModeHash RetVal;
+    RetVal.Hash = Value & 0x00FFFFFFFFFFFFFF;
+    RetVal.IsAuto = (Value & 0x2000000000000000) != 0;
+    RetVal.IsTest = (Value & 0x4000000000000000) != 0;
+    RetVal.IsEnabled = (Value & 0x8000000000000000) != 0;
+    return RetVal;
   }
 
   uint64_t ToValue() const {
-    uint64_t RetVal = Hash & 0x3FFFFFFFFFFFFFFF;
-    RetVal |= Enabled ? 0 : 0x8000000000000000;
-    RetVal |= NoCurrentOpMpode ? 0x4000000000000000 : 0;
+    uint64_t RetVal = Hash & 0x00FFFFFFFFFFFFFF;
+    RetVal |= IsAuto ? 0x2000000000000000 : 0;
+    RetVal |= IsTest ? 0x4000000000000000 : 0;
+    RetVal |= IsEnabled ? 0x8000000000000000 : 0;
```

Comment:
Do these constants need a `ULL` suffix?

Reply:
They didn't seem to, because these are not actually using the top bit.

Reply:
Isn't `0x8000'0000'0000'0000` the top bit of a `uint64_t`? It's equal to `1 << 63`, which is the top bit (remember that the LSB is `0x01 == 1 << 0`).